flink-clients/src/main/java/org/apache/flink/client/ClientUtils.java

```diff
@@ -128,16 +128,23 @@ public static void executeProgram(
 
 			LOG.info("Starting program (detached: {})", !configuration.getBoolean(DeploymentOptions.ATTACHED));
 
-			ContextEnvironmentFactory factory = new ContextEnvironmentFactory(
+			ContextEnvironment benv = new ContextEnvironment(
```

Comment:
This changes the behaviour quite a bit? Before, every call to `ExecutionEnvironment.getEnvironment()` would create a new env from the factory. Now we only have one environment and keep returning that from the anonymous inner factory created in `setAsContext()`. Whats the reason for this change?

Reply:
I forget user could call ExecutionEnvironment multiple times. Generally speaking these envs have same configuration but if we then modify them in user code they should changed.

The motivation I do this change is from `OptimizerPlanEnvironment` where return identical instance of env for all `getEnvironment` calls. However, `StreamPlanEnvironment` is different instance.

As the semantic of `getEnvironment`, I think return different instance for each call is reasonable.

Will update.

Reply:
Updated.